src/lib.rs

```diff
@@ -5774,6 +5774,11 @@ unsafe impl<'a> SplitByteSlice for RefMut<'a, [u8]> {
     }
 }
 
+// TODO(#429): Add a "SAFETY" comment and remove this `allow`.
+#[allow(clippy::undocumented_unsafe_blocks)]
+#[cfg(any(feature = "alloc", test))]
+unsafe impl ByteSlice for Vec<u8> {}
+
 // TODO(#429): Add a "SAFETY" comment and remove this `allow`.
 #[allow(clippy::undocumented_unsafe_blocks)]
 unsafe impl<'a> ByteSliceMut for &'a mut [u8] {}
```

Comment:
@joshlf, in light of the above comment, I'm not certain that this satisfies our documented safety condition either. It's trivial to define a function that changes both the length and address of a `&mut [u8]` between calls to `deref`:
```rust
fn change_add_and_length(mut slice: &mut [u8]) {
    slice = &mut slice[1..];
}
```
I think this indicates that our safety documentation on `ByteSlice` is missing some critical nuance.